rlog/state-separation.mdx

```diff
@@ -0,0 +1,1255 @@
+  - **Deshielded Execution**: Hides the sender's identity.
+- **zkVM Requirement**: If the execution involves private, shielded, or deshielded types, the user must call the zkVM to handle these confidential transactions. 
+For purely public executions, the zkVM is not needed, and the user can directly transmit the transaction code to the sequencer.
+
+**Step 3**: **Calling zkVM for Proof Generation**
+- **zkVM Compilation**: The user calls the zkVM to compile the smart contract with both public and private inputs.
+  - **Kernel Circuit Proofs**: The zkVM generates individual proofs for each execution type through kernel circuits.
+  - **Proof Aggregation**: The zkVM aggregates these individual proofs into a single comprehensive proof, combining both private and public inputs.
+
+**Step 4**: **Transmitting Public Inputs and Retaining Private Inputs**
+- **Retaining Private Inputs**: The user keeps the private inputs secure and does not transmit them.
+- **Revealing Public Inputs**: The user transmits the following public inputs to the sequencer:
+  - Public inputs of the recursive proof
+  - Hashes of UTXOs
+  - Updates to the public state
+  - Transaction signature
+  - Nullifiers (to prevent double spending)
+
+After completing these steps, the user's part of the execution is done, and the sequencer takes over the process.
+
+#### Sequencer Actions:
+
+**Step 5**: **Proof Verification**
+- **Proof and Data Reception**: The sequencer receives the proof and public inputs from the user.
+- **Verification Process**:
+  - For private, shielded, and deshielded executions, the sequencer verifies the proof using the provided public data.
+  - For public executions, the sequencer reruns the smart contract code with the provided inputs to check the results.
+- **Validation**: If both the zkVM proofs and public execution results are verified successfully, the sequencer collects the proof and public data to proceed. 
+If verification fails, the process is aborted, and the transaction is rejected.
+
+**Step 6**: **Aggregating Proofs and Finalizing the Block**
+- **Proof Aggregation**: The sequencer calls the zkVM again to aggregate all received proofs into one comprehensive proof to finalize the block.
+- **Finalizing the Block**:
+  - **Public State Update**: The sequencer updates the public state with the new transaction data.
+  - **Nullifier Tree Update**: Updates the nullifier tree to reflect the new state and prevent double spending.
+  - **Synchronization Mechanism**: Runs synchronization mechanisms to ensure fairness and consistency across the network.
+  - **UTXO Validation**: Validates the exchanged UTXOs to complete the transaction process.
+
+This comprehensive process ensures that transactions are executed securely, with the appropriate level of privacy and state updates synchronized across the network.
+
+In what follows, the execution process of the four different execution types within the Nescience architecture:
```

Comment:
The phrasing of this line is unnatural. I suggest something like:

"Below, we outline the execution process of the four different execution types within the Nescience architecture:"